pkg/targets/adapter/mongodbtarget/adapter.go

```diff
@@ -0,0 +1,250 @@
+/*
+Copyright 2023 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package mongodbtarget implements an adapter that connects to a MongoDB database
+// and allows a user to insert, query, and update documents via cloudevents.
+package mongodbtarget
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	pkgadapter "knative.dev/eventing/pkg/adapter/v2"
+	"knative.dev/pkg/logging"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/zap"
+
+	"github.com/triggermesh/triggermesh/pkg/apis/targets"
+	"github.com/triggermesh/triggermesh/pkg/apis/targets/v1alpha1"
+	"github.com/triggermesh/triggermesh/pkg/metrics"
+	targetce "github.com/triggermesh/triggermesh/pkg/targets/adapter/cloudevents"
+)
+
+// NewTarget returns the adapter implementation.
+func NewTarget(ctx context.Context, envAcc pkgadapter.EnvConfigAccessor, ceClient cloudevents.Client) pkgadapter.Adapter {
+	env := envAcc.(*envAccessor)
+	logger := logging.FromContext(ctx)
+	metrics.MustRegisterEventProcessingStatsView()
+
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(env.ServerURL))
+	if err != nil {
+		logger.Fatal("error connecting to mongodb", err)
```

Comment:
maybe this is a bit more friendly for log parsers

```suggestion
		logger.Panicw("error connecting to mongodb", zap.Error(err))
```

Reply:
xd :) good catch